pelita/game.py

```diff
@@ -785,16 +792,18 @@ def check_gameover(game_state, detect_final_move=False):
             if num_errors[team] > MAX_ALLOWED_ERRORS:
                 return { 'whowins' : 1 - team, 'gameover' : True}
 
-    if detect_final_move:
+    if detect_final_move and game_state['round'] is not None:
         # No team wins/loses because of errors?
         # Good. Now check if the game finishes because the food is gone
         # or because we are in the final turn of the last round.
-        next_step = update_round_counter(game_state)
-        _logger.debug(f"Next step has {next_step}")
+        #next_step = update_round_counter(game_state)
+        next_round = game_state['round']
```

Comment:
because it was using `update_round_counter` for its side effects...

Reply:
I guess `update_round_counter` should be renamed to `get_next_round_turn` then, as it is not a side-effect but the only thing that it does. :)

Reply:
But the game is finishing correctly at the right round even without this function call. Why was it there?

Reply:
Of course it is working now, because you reimplemented the core of the function and calculate the next round manually now. In the spirit of code-reuse we might want to revert this.

Reply:
what about getting rid of the function alltogether? It is one of those functions where we asked why they exist at all.

Reply:
This is why it exists! ![image](https://user-images.githubusercontent.com/216179/61721883-908f6a00-ad69-11e9-8469-977eee1ebfef.png)